pkg/migrations/sql/idempotent/telemetry.sql

```diff
@@ -0,0 +1,94 @@
+            promscale_trace_dependency_requests_executed_total  = subquery.counter_reset_traces_dependency_queries_executed + subquery.stale_traces_dependency_queries_executed
+        FROM
+            (
+                -- Basically, we are collecting all stale rows under `stale.` and the actual counter-reset row under `counter_reset.`
+                -- and the are aggregated in the above SET step. We can aggregate it here, but its more cleaner to do in SET.
+                SELECT
+                    sum(stale.promscale_ingested_samples_total)  			        stale_metrics_ingested_samples,
+                    sum(stale.promscale_metrics_queries_executed_total)  		    stale_metrics_queries_executed,
+                    sum(stale.promscale_metrics_queries_timedout_total)  			stale_metrics_queries_timed_out,
+                    sum(stale.promscale_metrics_queries_failed_total)  				stale_metrics_queries_failed,
+                    sum(stale.promscale_trace_query_requests_executed_total)        stale_traces_queries_executed,
+                    sum(stale.promscale_trace_dependency_requests_executed_total)  	stale_traces_dependency_queries_executed,
+
+                    counter_reset.promscale_ingested_samples_total  			        counter_reset_metrics_ingested_samples,
+                    counter_reset.promscale_metrics_queries_executed_total 			    counter_reset_metrics_queries_executed,
+                    counter_reset.promscale_metrics_queries_timedout_total  			counter_reset_metrics_queries_timed_out,
+                    counter_reset.promscale_metrics_queries_failed_total  			    counter_reset_metrics_queries_failed,
+                    counter_reset.promscale_trace_query_requests_executed_total  	    counter_reset_traces_queries_executed,
+                    counter_reset.promscale_trace_dependency_requests_executed_total    counter_reset_traces_dependency_queries_executed
+                FROM
+                    SCHEMA_PS_CATALOG.promscale_instance_information stale INNER JOIN SCHEMA_PS_CATALOG.promscale_instance_information counter_reset ON true
+                WHERE 
+                    counter_reset.is_counter_reset_row = TRUE AND (stale.is_counter_reset_row = FALSE AND current_timestamp - stale.last_updated > interval '2 HOUR') -- consider only those rows as stale who were last updated beyond 2 hours.
+                GROUP BY
+                    counter_reset.promscale_ingested_samples_total,
+                    counter_reset.promscale_metrics_queries_executed_total,
+                    counter_reset.promscale_metrics_queries_timedout_total,
+                    counter_reset.promscale_metrics_queries_failed_total,
+                    counter_reset.promscale_trace_query_requests_executed_total,
+                    counter_reset.promscale_trace_dependency_requests_executed_total
+            ) AS subquery
+        WHERE is_counter_reset_row = TRUE;
```

Comment:
simplify this update statement

Reply:
```
UPDATE SCHEMA_PS_CATALOG.promscale_instance_information SET
    promscale_ingested_samples_total                    = x.promscale_ingested_samples_total,
    promscale_metrics_queries_executed_total            = x.promscale_metrics_queries_executed_total,
    promscale_metrics_queries_timedout_total            = x.promscale_metrics_queries_timedout_total,
    promscale_metrics_queries_failed_total              = x.promscale_metrics_queries_failed_total,
    promscale_trace_query_requests_executed_total       = x.promscale_trace_query_requests_executed_total,
    promscale_trace_dependency_requests_executed_total  = x.promscale_trace_dependency_requests_executed_total,
    last_updated = current_timestamp
FROM
(
    SELECT
        sum(promscale_ingested_samples_total)  			        as promscale_ingested_samples_total,
        sum(promscale_metrics_queries_executed_total)  		    as promscale_metrics_queries_executed_total,
        sum(promscale_metrics_queries_timedout_total)  		    as promscale_metrics_queries_timedout_total,
        sum(promscale_metrics_queries_failed_total)  		    as promscale_metrics_queries_failed_total,
        sum(promscale_trace_query_requests_executed_total)      as promscale_trace_query_requests_executed_total,
        sum(promscale_trace_dependency_requests_executed_total) as promscale_trace_dependency_requests_executed_total
    FROM
        SCHEMA_PS_CATALOG.promscale_instance_information
    WHERE 
        is_counter_reset_row = TRUE
        OR (is_counter_reset_row = FALSE AND current_timestamp - last_updated > interval '1 HOUR') -- consider only those rows as stale who were last updated beyond an hour.
) x
WHERE is_counter_reset_row = TRUE;
```

Reply:
```
WITH deleted_rows AS (
    DELETE FROM SCHEMA_PS_CATALOG.promscale_instance_information 
    WHERE is_counter_reset_row = FALSE AND current_timestamp - last_updated > interval '2 HOUR'
    RETURNING *
)
UPDATE SCHEMA_PS_CATALOG.promscale_instance_information SET
    promscale_ingested_samples_total                    = promscale_ingested_samples_total + x.del_promscale_ingested_samples_total,
    promscale_metrics_queries_executed_total            = promscale_metrics_queries_executed_total + x.del_promscale_metrics_queries_executed_total,
    promscale_metrics_queries_timedout_total            = promscale_metrics_queries_timedout_total + x.del_promscale_metrics_queries_timedout_total,
    promscale_metrics_queries_failed_total              = promscale_metrics_queries_failed_total + x.del_promscale_metrics_queries_failed_total,
    promscale_trace_query_requests_executed_total       = promscale_trace_query_requests_executed_total + x.del_promscale_trace_query_requests_executed_total,
    promscale_trace_dependency_requests_executed_total  = promscale_trace_dependency_requests_executed_total + x.del_promscale_trace_dependency_requests_executed_total,
    last_updated = current_timestamp
FROM
(
    SELECT
        sum(promscale_ingested_samples_total)  			        as del_promscale_ingested_samples_total,
        sum(promscale_metrics_queries_executed_total)  		    as del_promscale_metrics_queries_executed_total,
        sum(promscale_metrics_queries_timedout_total)  		    as del_promscale_metrics_queries_timedout_total,
        sum(promscale_metrics_queries_failed_total)  		    as del_promscale_metrics_queries_failed_total,
        sum(promscale_trace_query_requests_executed_total)      as del_promscale_trace_query_requests_executed_total,
        sum(promscale_trace_dependency_requests_executed_total) as del_promscale_trace_dependency_requests_executed_total
    FROM
        deleted_rows
) x
WHERE is_counter_reset_row = TRUE;
```

Reply:
with the above you can get rid of line 89 as well as the check on line 40 and the extra update on line 35

Reply:
Why do we need to know the number of deleted rows?

If it's fine, I will love to keep this simple, which is the most recent pushed version. (which John has suggested)